### PR TITLE
Add the unfollow user endpoint

### PR DESCRIPTION
**Pull request by romiparada**
https://trello.com/c/4kNTms5N

### CODE REVIEW

File: app/controllers/api/users_controller.rb [hunk 1]
```diff
@@ -18,6 +18,12 @@ def follow
       render status: :no_content
     end
 
+    def unfollow
+      user.follower_users.find(current_user.id)
+      user.follower_users.destroy(current_user)
```
Comment (juan-apa) [inline app/controllers/api/users_controller.rb:23]: maybe refactor this to:
```ruby
current_user.followings.find_by!(following: user).destroy!
```

File: spec/requests/api/users/unfollow_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      it 'returns 204 status code' do
+        subject
+        expect(response).to have_http_status(:no_content)
+      end
+
+      it 'deletes follow' do
+        subject
+        expect(following.follower_users).not_to include(user)
+        expect(user.following_users).not_to include(following)
+      end
```
Comment (juan-apa) [inline spec/requests/api/users/unfollow_spec.rb:26]:
```ruby
expect { subject }.to change { follow }.from(1).to(0)
expect { follow.reload }.to raise(ActiveRecord::RecordNotFound)
```

File: spec/requests/api/users/unfollow_spec.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  subject { post unfollow_user_path(username), headers:, as: :json }
+
+  let(:user) { create(:user) }
+  let(:headers) { auth_headers(user) }
+  let(:following) { create(:user) }
+  let(:username) { following.username }
+
+  context 'when the params are correct' do
+    context 'when the user unfollows a following user' do
+      before { user.following_users << following }
```
Comment (juan-apa) [inline spec/requests/api/users/unfollow_spec.rb:15]:
```ruby
let!(:follow) { create(:follow, follower: user, following:) }
```

**Review by juan-apa — APPROVED** [review]

File: app/controllers/api/users_controller.rb (hunk 1 above)
Comment (romiparada) [reply app/controllers/api/users_controller.rb:23]: What about `current_user.following_users.find(user.id).destroy!`, to use the `has_many through` association

**Pull request merged**